### DISCUSSION TITLE
Difference in Clearsky DNI Results Between PVLib and Solcast

### DISCUSSION THREAD

**Discussion by HrvojeR** [pvlib/pvlib-python#2297] [Q&A] [answered]
Dear friends,

I am using the PVLib library to calculate clearsky DNI values on an hourly basis for a specific geographic location. However, I have noticed significant differences between the results obtained using the PVLib method and the values provided by the Solcast API. To test the accuracy of PVLib methods, I used data from Solcast as input for PVLib. The codei s shown below:

```
import pandas as pd
from pvlib.location import Location

# Loading Solcast data
data = pd.read_excel('data.xlsx', parse_dates=['period_start'], index_col='period_start')

# Defining geographic location
latitude = _my location data_
longitude = _my location data_

location = Location(latitude, longitude, 'Europe/Zagreb')

# Solar position - data from Solcast
solar_position = pd.DataFrame(
    {
        'apparent_zenith': data['zenith'],
        'zenith': data['zenith'],
        'apparent_elevation': data['apparent_elevation']
    },
    index=data.index
)

# Calculating clearsky values using PVLib
clearsky = location.get_clearsky(
    times=data.index, 
    model='ineichen', 
    solar_position=solar_position, 
    pressure=data['pressure']
)

# Obtained Clearsky DNI values
print(clearsky['dni'])

```
For comparison, here are the clearsky DNI results obtained using PVLib and the values returned by the Solcast API:

PVLib Clearsky DNI:

```
pvlib clearsky_dni:
0
0
0
0
0
0
0
190.459769
504.166109
657.054597
727.709442
750.209492
727.709442
673.140539
556.541914
285.565017
0
0
0
0
0
0
0
0
solcast clearsky_dni:

0
0
0
0
0
0
2
382
683
795
839
849
837
785
666
397
13
0
0
0
0
0
0
0
```
The differences are evident, particularly during periods of high solar irradiance. My questions are:

1. What could be the reasons for these differences?
2. How can I adjust PVLib settings (e.g., model selection, additional parameters) to align the results more closely with those provided by Solcast?
3. Is it possible that Solcast applies a specific model or correction that PVLib does not implement?

Thank you in advance for your assistance.

Best regards,

**Comment by cwhanse** [comment]
It's difficult to comment when location, time, and SolCast data are omitted. There may be differences due to the time resolution and alignment: values from pvlib are instantaneous at the timestamp; SolCast may be an average or other statistic for the interval (I don't know). Also, the comparison is between two models one of which isn't public: https://www.solcast.com/irradiance-data-methodology

**Comment by mikofski** [comment]
In addition to input differences, please note there are several clear sky models. The default in pvlib is “Ineichen” but we don’t know what clear sky model Solcast uses. Please see [pvlib clear sky documentation](https://pvlib-python.readthedocs.io/en/stable/reference/clearsky.html) for a list of models implemented in pvlib, & note there may be other models.

**Reply by wholmgren** [reply]
Solcast uses REST2 + modifications. https://www.solcast.com/irradiance-data-methodology

Text below is from [DNV review of Solcast historical data](https://www.dnv.com/publications/review-and-validation-of-solcast-historical-and-tmy-data/)

![image](https://github.com/user-attachments/assets/7ac53374-bbde-4825-a118-5e91478fb7a3)

**Reply by adriesse** [reply]
Are the extensive validations of REST2 valid after "proprietary configurations and improvements"??

**Comment by AdamRJensen** [comment, marked as answer]
Solcast is based on actual current aerosol data, whereas the pvlib function you use is based on climatological average aerosol data (high uncertainty).

I would recommend comparing against the free McClear clear sky data (this can be retrieved using the pvlib.iotools.get_cams function).

**Comment by HrvojeR** [comment]
Thank you all for your responses; they have been extremely helpful. Guided by the suggestion @AdamRJensen  which we tested, we achieved the most accurate results so far, and we hope they will prove useful in practice. Once again, thank you all!
Results:
https://prnt.sc/qCKBmrQuVSMa
